apps/judicial-system/web/src/routes/Shared/DetentionRequests/utils.tsx

```diff
@@ -0,0 +1,57 @@
+import { TagVariant } from '@island.is/island-ui/core'
+import { SortConfig } from '@island.is/judicial-system-web/src/types'
+import { Case, CaseState, UserRole } from '@island.is/judicial-system/types'
+import router from 'next/router'
+import * as Constants from '@island.is/judicial-system-web/src/utils/constants'
+
+export const mapCaseStateToTagVariant = (
+  state: CaseState,
+  isJudge: boolean,
+  isCustodyEndDateInThePast?: boolean,
+): { color: TagVariant; text: string } => {
+  switch (state) {
+    case CaseState.NEW:
+    case CaseState.DRAFT:
+      return { color: 'red', text: 'Drög' }
+    case CaseState.SUBMITTED:
+      return {
+        color: 'purple',
+        text: `${isJudge ? 'Ný krafa' : 'Krafa send'}`,
+      }
+    case CaseState.RECEIVED:
+      return { color: 'darkerMint', text: 'Krafa móttekin' }
+    case CaseState.ACCEPTED:
+      if (isCustodyEndDateInThePast) {
+        return {
+          color: 'darkerBlue',
+          text: 'Lokið',
+        }
+      } else {
+        return {
+          color: 'blue',
+          text: 'Virkt',
+        }
+      }
+    case CaseState.REJECTED:
+      return { color: 'rose', text: 'Kröfu hafnað' }
+    default:
+      return { color: 'white', text: 'Óþekkt' }
+  }
+}
+
+export const handleClick = (
```

Comment:
Þetta function ætti að heita, go to case eða eitthvað álíka. mögulega að taka in case ekki parta af case.